clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -1011,6 +1011,25 @@ public void onComplete(Map<TopicPartition, OffsetAndMetadata> offsets, Exception
         }
     }
 
+    private RequestFuture<Void> sendOffsetCommitRequestAndRefreshHeartbeat(final Map<TopicPartition, OffsetAndMetadata> offsets) {
```

Comment:
Instead of wrapping `sendOffsetCommitRequest` with another function, I'm wondering if we should just augment `sendOffsetCommitRequest` itself to take the heartbeat update as well, if we consider each offset commit to be treated as a heartbeat too?

Reply:
I agree that sounds simpler. Note also that we'll want to call `Heartbeat.failHeartbeat` in the case of an offset commit failure.

Reply:
Yes, it's better to augment `sendOffsetCommitRequest` method, I changed it, take a look @hachikuji